### PR TITLE
aks/use-wasi-node-pools: update the doc to refer to SpinKube

### PR DESCRIPTION
**Pull request by Mossaka**
* replace the slight workload with spin workload
* add a section to describe all the supported runtime classes
* few changes to the wording

### CODE REVIEW

File: articles/aks/use-wasi-node-pools.md [hunk 1]
```diff
@@ -58,15 +58,15 @@ az provider register --namespace Microsoft.ContainerService
 
 ## Limitations
 
-* Currently, there are only containerd shims available for [spin][spin] and [slight][slight] applications, which use the [wasmtime][wasmtime] runtime. In addition to wasmtime runtime applications, you can also run containers on WASM/WASI node pools.
-* You can run containers and wasm modules on the same node, but you can't run containers and wasm modules on the same pod.
```
Comment (Mossaka) [inline articles/aks/use-wasi-node-pools.md:62]: you can actually run Linux and Wasm containers side-by-side with the latest version now!

File: articles/aks/use-wasi-node-pools.md [hunk 2]
```diff
@@ -58,15 +58,15 @@ az provider register --namespace Microsoft.ContainerService
 
 ## Limitations
 
-* Currently, there are only containerd shims available for [spin][spin] and [slight][slight] applications, which use the [wasmtime][wasmtime] runtime. In addition to wasmtime runtime applications, you can also run containers on WASM/WASI node pools.
```
Comment (Mossaka) [inline articles/aks/use-wasi-node-pools.md:61]: slight is deprecated, so I've decided to remove the mentioning of slight completely

File: articles/aks/use-wasi-node-pools.md [hunk 3]
```diff
@@ -9,12 +9,12 @@ ms.author: schaffererin
...
+[WebAssembly (Wasm)][wasm] is a binary format that is optimized for fast download and maximum execution speed in a Wasm runtime. A Wasm runtime is designed to run on a target architecture and execute Wasm components in a sandbox, isolated from the host computer, at near-native performance. By default, Wasm components can't access resources on the host outside of the sandbox unless it is explicitly allowed, and they can't communicate over sockets to access things like environment variables or HTTP traffic. The [WebAssembly System Interface (WASI)][wasi] standard defines an API for Wasm runtimes to provide access to Wasm components to the environment and resources outside the host using a capabilities model.
```
Comment (devigned) [inline articles/aks/use-wasi-node-pools.md:14]:
```suggestion
[WebAssembly (Wasm)][wasm] is a binary format that is optimized for fast download and maximum execution speed in a Wasm runtime. A Wasm runtime is designed to run on a target architecture and execute Wasm components in a sandbox, isolated from the host computer, at near-native performance. By default, Wasm components can't access resources on the host outside of the sandbox unless it is explicitly allowed. For example, Wasm components can't communicate over sockets, receive or send HTTP traffic, or even to access things like environment variables without explicit approval. The [WebAssembly System Interface (WASI)][wasi] standard defines an API for Wasm runtimes to provide access to Wasm components to the environment and resources outside the host using a capabilities model.
```

File: articles/aks/use-wasi-node-pools.md [hunk 4]
```diff
@@ -9,12 +9,12 @@ ms.author: schaffererin
...
 
 > [!IMPORTANT]
-> WASI nodepools now use [containerd shims][wasm-containerd-shims] to run WASM workloads. Previously, AKS used [Krustlet][krustlet] to allow WASM modules to be run on Kubernetes. If you are still using Krustlet-based WASI nodepools, you can migrate to containerd shims by creating a new WASI nodepool and migrating your workloads to the new nodepool.
+> WASI nodepools now use [SpinKube shim][containerd-shim-spin] to run Wasm workloads. Previously, AKS used [Krustlet][krustlet] to allow Wasm modules to be run on Kubernetes. If you are still using Krustlet-based WASI nodepools, you can migrate to containerd shims by creating a new WASI nodepool and migrating your workloads to the new nodepool.
```
Comment (devigned) [inline articles/aks/use-wasi-node-pools.md:17]:
```suggestion
> WASI nodepools now use [SpinKube containerd shim][containerd-shim-spin] to run Wasm workloads. Previously, AKS used [Krustlet][krustlet] to allow Wasm modules to be run on Kubernetes. If you are still using Krustlet-based WASI nodepools, you can migrate to containerd shims by creating a new WASI nodepool and migrating your workloads to the new nodepool.
```

File: articles/aks/use-wasi-node-pools.md [hunk 5]
```diff
@@ -58,15 +58,15 @@ az provider register --namespace Microsoft.ContainerService
 
 ## Limitations
 
-* Currently, there are only containerd shims available for [spin][spin] and [slight][slight] applications, which use the [wasmtime][wasmtime] runtime. In addition to wasmtime runtime applications, you can also run containers on WASM/WASI node pools.
-* You can run containers and wasm modules on the same node, but you can't run containers and wasm modules on the same pod.
-* The WASM/WASI node pools can't be used for system node pool.
-* The *os-type* for WASM/WASI node pools must be Linux.
-* You can't use the Azure portal to create WASM/WASI node pools.
+* Currently, there are only containerd shims available for [spin][spin] applications, which use the [wasmtime][wasmtime] runtime. In addition to wasmtime runtime applications, you can also run containers on Wasm/WASI node pools.
```
Comment (devigned) [inline articles/aks/use-wasi-node-pools.md:61]:
```suggestion
* Currently, WASI Node Pools only support [Spin][spin] applications which use the [Wasmtime][wasmtime] runtime, and native Linux containers.
```

**Review by devigned — CHANGES_REQUESTED** [review]
Looks good. I proposed a few places for adjustment. Please consider the suggestions.

**Comment by Jak-MS** [comment]
@schaffererin 
Can you review the proposed changes? 

**Important**: When the changes are ready for publication, adding a `#sign-off` comment is the best way to signal that the PR is ready for the review team to merge. 

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

File: articles/aks/use-wasi-node-pools.md [hunk 6]
```diff
@@ -58,15 +58,15 @@ az provider register --namespace Microsoft.ContainerService
 
 ## Limitations
 
-* Currently, there are only containerd shims available for [spin][spin] and [slight][slight] applications, which use the [wasmtime][wasmtime] runtime. In addition to wasmtime runtime applications, you can also run containers on WASM/WASI node pools.
-* You can run containers and wasm modules on the same node, but you can't run containers and wasm modules on the same pod.
-* The WASM/WASI node pools can't be used for system node pool.
-* The *os-type* for WASM/WASI node pools must be Linux.
-* You can't use the Azure portal to create WASM/WASI node pools.
+* Currently, WASI Node Pools only support [Spin][spin] applications which use the [Wasmtime][wasmtime] runtime, and native Linux containers.
```
Comment (devigned) [inline articles/aks/use-wasi-node-pools.md:61]:
```suggestion
* Currently, WASI Node Pools only support [Spin][spin] applications, which use the [Wasmtime][wasmtime] runtime and native Linux containers.
```

I'm sorry. This was from my feedback. I think you do you need the comma in front of which.

File: articles/aks/use-wasi-node-pools.md [hunk 7]
```diff
@@ -58,15 +58,15 @@ az provider register --namespace Microsoft.ContainerService
 
 ## Limitations
 
-* Currently, there are only containerd shims available for [spin][spin] and [slight][slight] applications, which use the [wasmtime][wasmtime] runtime. In addition to wasmtime runtime applications, you can also run containers on WASM/WASI node pools.
-* You can run containers and wasm modules on the same node, but you can't run containers and wasm modules on the same pod.
-* The WASM/WASI node pools can't be used for system node pool.
-* The *os-type* for WASM/WASI node pools must be Linux.
-* You can't use the Azure portal to create WASM/WASI node pools.
+* Currently, WASI Node Pools only support [Spin][spin] applications, which use the [Wasmtime][wasmtime] runtime and native Linux containers.
```
Comment (schaffererin) [inline articles/aks/use-wasi-node-pools.md:61]:
```suggestion
* Currently, WASI node pools only support [Spin][spin] applications, which use the [Wasmtime][wasmtime] runtime and native Linux containers.
```

**Review by schaffererin — CHANGES_REQUESTED** [review]
Small nit, otherwise it LGTM!

File: articles/aks/use-wasi-node-pools.md (hunk 7 above)
Comment (schaffererin) [reply articles/aks/use-wasi-node-pools.md:61]: It also looks like there might be an invalid file link. Once you make those two changes, please let me know and I can sign off on the PR @Mossaka

Comment (Mossaka) [reply articles/aks/use-wasi-node-pools.md]: Resolved

**Comment by Mossaka** [comment]
I've resolved all the comments. Could you please take another look? @schaffererin

**Comment by schaffererin** [comment]
#sign-off

**Pull request merged**